src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,133 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int FIRST_BUCKET_INDEX = 0;
+    private Node<K, V>[] innerArray;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        innerArray = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (LOAD_FACTOR * DEFAULT_CAPACITY);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int targetBucket = findTargetBucketByKey(key);
+        Node<K, V> newNode = createNewNode(key, value);
+        if (innerArray[targetBucket] == null) {
+            innerArray[targetBucket] = newNode;
+        } else {
+            Node<K, V> lastNodeInBucket = innerArray[targetBucket];
+            Node<K, V> preLastNodeInBucket = innerArray[targetBucket];
+            while (lastNodeInBucket != null) {
+                if (lastNodeInBucket.key == key || lastNodeInBucket.key != null
+                        && lastNodeInBucket.key.equals(key)) {
+                    lastNodeInBucket.value = value;
+                    return;
+                }
+                preLastNodeInBucket = lastNodeInBucket;
+                lastNodeInBucket = lastNodeInBucket.next;
+            }
+            preLastNodeInBucket.next = newNode;
```

Comment:
U can simplify this code if u add in 'while' check  'lastNodeInBucket.next != null' (u check that  'innerArray[targetBucket]' isn`t null in previous case), it gives u opportunity to remove 'preLastNodeInBucket'